File: src/main/java/com/enderryno/nuclearcraft/enums/ItemBehaviour.java
package com.enderryno.nuclearcraft.enums;

public enum ItemBehaviour {
    BLOCK("block"),
    GENERIC("generic"),
    GAS_MASK("gas_mask"),
    GAS_MASK_FILTER("gas_mask_filter"),
    RADIATION_INHIBITOR("radiation_inhibitor"),
    GEIGER_COUNTER("geiger_counter");

    final String name;
    ItemBehaviour(String name) {
        this.name = name;
    }
    public static ItemBehaviour fromString(String text) {
        for (ItemBehaviour b : ItemBehaviour.values()) {
            if (b.name.equalsIgnoreCase(text)) {
                return b;
            }
        }
        return null;
    }

    public String getName() {
        return name;
    }
}

File: src/main/java/com/enderryno/nuclearcraft/enums/ItemBehaviour.kt
package com.enderryno.nuclearcraft.enums

enum class ItemBehaviour(name: String) {
    BLOCK("block"),
    GENERIC("generic"),
    GAS_MASK("gas_mask"),
    GAS_MASK_FILTER("gas_mask_filter"),
    RADIATION_INHIBITOR("radiation_inhibitor"),
    GEIGER_COUNTER("geiger_counter");

    companion object {
        fun fromString(text: String?): ItemBehaviour? {
            for (b in values()) {
                if (b.name.equals(text, ignoreCase = true)) {
                    return b
                }
            }
            return null
        }
    }
}
